crates/transaction-generator-lib/src/reduced_batch_wrapper.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{TransactionGenerator, TransactionGeneratorCreator};
+use aptos_sdk::types::{transaction::SignedTransaction, LocalAccount};
+
+struct ReducedBatchWrapperTransactionGenerator {
+    batch_size: usize,
+    generator: Box<dyn TransactionGenerator>,
+}
+
+impl TransactionGenerator for ReducedBatchWrapperTransactionGenerator {
+    fn generate_transactions(
+        &mut self,
+        account: &LocalAccount,
+        num_to_create: usize,
+    ) -> Vec<SignedTransaction> {
+        self.generator
+            .generate_transactions(account, num_to_create.min(self.batch_size))
```

Comment:
nit: So technically this is not always reduced. It is just capped/limited/bounded. Is "BoundedBatchWrapperTransactionGenerator" a better name ?